File: app/src/main/java/com/bas/google_book_app/db/BookPreferences.java
package com.bas.google_book_app.db;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.bas.google_book_app.R;

public class BookPreferences {

    public static String getFilterPreference(Context context) {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
        String filterPrefKey = context.getString(R.string.pref_key_filter_by);
        String filterDefault = context.getString(R.string.pref_filter_by_default);

        return preferences.getString(filterPrefKey, filterDefault);
    }
}

File: app/src/main/java/com/bas/google_book_app/db/BookPreferences.kt
package com.bas.google_book_app.db

import android.content.Context
import android.preference.PreferenceManager
import com.bas.google_book_app.R

object BookPreferences {
    fun getFilterPreference(context: Context?): String? {
        val preferences = PreferenceManager.getDefaultSharedPreferences(context)
        val filterPrefKey = context?.getString(R.string.pref_key_filter_by)
        val filterDefault = context?.getString(R.string.pref_filter_by_default)
        return preferences.getString(filterPrefKey, filterDefault)
    }
}